pkg/cmd/release/view/view.go

```diff
@@ -217,3 +310,104 @@ func floatToString(f float64, p uint8) string {
 	idx := strings.IndexRune(fs, '.')
 	return fs[:idx+int(p)+1]
 }
+
+func matchTag(pattern, tag string) (bool, error) {
+	g, err := glob.Compile(pattern)
+	if err != nil {
+		return false, err
+	}
+
+	return g.Match(tag), nil
+}
+
+type Range interface {
+	Match(string) bool
+}
+
+type Interval struct {
+	start string
+	end   string
+}
+
+func NewInterval(start, end string) (*Interval, error) {
+	if !semver.IsValid(start) {
+		return nil, &ErrInvalidSemver{Semver: start}
+	} else if !semver.IsValid(semver.Canonical(end)) {
+		return nil, &ErrInvalidSemver{Semver: end}
+	}
+
+	// protecting myself from stupid errors
+	if semver.Compare(start, end) == 1 {
+		return &Interval{
+			start: end,
+			end:   start,
+		}, nil
+	}
+
+	return &Interval{
+		start: start,
+		end:   end,
+	}, nil
+}
+
+func (i Interval) Match(tag string) bool {
+	if semver.Compare(i.start, tag) > 0 || semver.Compare(i.end, tag) < 0 {
+		return false
+	} else {
+		return true
+	}
+}
+
+type Glob struct {
+	glob glob.Glob
+}
+
+func NewGlob(pattern string) (*Glob, error) {
```

Comment:
Nitpick: types and functions that are only used within the same package should not be exported